File: frontend/types/vacancy.ts
export type VacancyStatus = "draft" | "approved" | "closed" | "archived";

export interface Vacancy {
  id: string;
  title: string | null;
  skills: string[] | null;
  area: string | null;
  area_hh_id: string | null;
  salary_from: number | null;
  salary_to: number | null;
  currency: string;
  experience: string | null;
  employment_type: string | null;
  schedule: string | null;
  description: string | null;
  status: VacancyStatus;
  is_open: boolean;
  is_approvable: boolean;
  last_matched_at: string | null;
  created_at: string;
  updated_at: string;
  score_criteria: { name: string; weight: number }[];
}

export interface AIFields {
  title?: string | null;
  skills?: string[] | null;
  area?: string | null;
  salary_from?: number | null;
  salary_to?: number | null;
  currency?: string | null;
  experience?: string | null;
  employment_type?: string | null;
  schedule?: string | null;
  description?: string | null;
}

export const EXPERIENCE_OPTIONS = [
  { value: "noExperience", label: "Без опыта" },
  { value: "between1And3", label: "1–3 года" },
  { value: "between3And6", label: "3–6 лет" },
  { value: "moreThan6", label: "6+ лет" },
];

export const EMPLOYMENT_OPTIONS = [
  { value: "full", label: "Полная занятость" },
  { value: "part", label: "Частичная занятость" },
  { value: "project", label: "Проектная работа" },
  { value: "volunteer", label: "Волонтёрство" },
  { value: "probation", label: "Стажировка" },
];

export const SCHEDULE_OPTIONS = [
  { value: "fullDay", label: "Полный день" },
  { value: "shift", label: "Сменный" },
  { value: "flexible", label: "Гибкий" },
  { value: "remote", label: "Удалённо" },
  { value: "flyInFlyOut", label: "Вахтовый метод" },
];

export const CURRENCY_OPTIONS = [
  { value: "UZS", label: "UZS" },
  { value: "USD", label: "$ USD" },
  { value: "RUB", label: "₽ RUB" },
  { value: "EUR", label: "€ EUR" },
];

File: backend/app/models/vacancy.py
from datetime import datetime
from enum import Enum
from typing import List, Optional

from beanie import Document
from pydantic import Field


class VacancyStatus(str, Enum):
    draft = "draft"
    approved = "approved"
    closed = "closed"
    archived = "archived"


class Vacancy(Document):
    title: Optional[str] = None
    skills: List[str] = []
    area: Optional[str] = None
    area_hh_id: Optional[str] = None
    salary_from: Optional[int] = None
    salary_to: Optional[int] = None
    currency: str = "UZS"
    experience: Optional[str] = None      # noExperience | between1And3 | between3And6 | moreThan6
    employment_type: Optional[str] = None # full | part | project | volunteer | probation
    schedule: Optional[str] = None        # fullDay | shift | flexible | remote | flyInFlyOut
    description: Optional[str] = None
    raw_description: Optional[str] = None
    status: VacancyStatus = VacancyStatus.draft
    score_criteria: List[dict] = Field(default_factory=lambda: [
        {"name": "Соответствие должности", "weight": 45},
        {"name": "Навыки и инструменты",   "weight": 40},
        {"name": "Уровень опыта",          "weight": 15},
    ])
    is_open: bool = True
    hh_vacancy_id: Optional[str] = None
    last_matched_at: Optional[datetime] = None
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

    def is_approvable(self) -> bool:
        required = [self.title, self.description, self.experience]
        return all(f is not None and f != "" for f in required)

    class Settings:
        name = "vacancies"
